runtime-modules/content/src/lib.rs

```diff
@@ -0,0 +1,1142 @@
+};
+#[cfg(feature = "std")]
+pub use serde::{Deserialize, Serialize};
+use sp_arithmetic::traits::{BaseArithmetic, One, Zero};
+use sp_runtime::traits::{MaybeSerializeDeserialize, Member};
+use sp_std::collections::btree_set::BTreeSet;
+// use sp_std::vec;
+use sp_std::vec::Vec;
+use system::ensure_signed;
+
+pub use common::storage::{ContentParameters, StorageSystem};
+pub use common::{
+    currency::{BalanceOf, GovernanceCurrency},
+    MembershipTypes, StorageOwnership,
+};
+
+pub(crate) type ContentId<T> = <T as StorageOwnership>::ContentId;
+
+pub(crate) type DataObjectTypeId<T> = <T as StorageOwnership>::DataObjectTypeId;
+
+/// Type, used in diffrent numeric constraints representations
+pub type MaxNumber = u32;
+
+/// A numeric identifier trait
+pub trait NumericIdentifier:
+    Parameter
+    + Member
+    + BaseArithmetic
+    + Codec
+    + Default
+    + Copy
+    + Clone
+    + Hash
+    + MaybeSerializeDeserialize
+    + Eq
+    + PartialEq
+    + Ord
+    + Zero
+{
+}
+
+impl NumericIdentifier for u64 {}
+
+/// Module configuration trait for Content Directory Module
+pub trait Trait:
+    system::Trait
+    + ContentActorAuthenticator
+    + Clone
+    + StorageOwnership
+    + MembershipTypes
+    + GovernanceCurrency
+{
+    /// The overarching event type.
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+    /// Channel Transfer Payments Escrow Account seed for ModuleId to compute deterministic AccountId
+    type ChannelOwnershipPaymentEscrowId: Get<[u8; 8]>;
+
+    /// ChannelRevenueTreasury seed for ModuleId to compute deterministic AccountId
+    type ChannelRevenueTreasuryId: Get<[u8; 8]>;
+
+    /// Type of identifier for Videos
+    type VideoId: NumericIdentifier;
+
+    /// Type of identifier for Video Categories
+    type VideoCategoryId: NumericIdentifier;
+
+    /// Type of identifier for Channel Categories
+    type ChannelCategoryId: NumericIdentifier;
+
+    /// Type of identifier for Playlists
+    type PlaylistId: NumericIdentifier;
+
+    /// Type of identifier for Persons
+    type PersonId: NumericIdentifier;
+
+    /// Type of identifier for Channels
+    type SeriesId: NumericIdentifier;
+
+    /// Type of identifier for Channel transfer requests
+    type ChannelOwnershipTransferRequestId: NumericIdentifier;
+
+    /// The maximum number of curators per group constraint
+    type MaxNumberOfCuratorsPerGroup: Get<MaxNumber>;
+
+    // Type that handles asset uploads to storage system
+    type StorageSystem: StorageSystem<Self>;
+}
+
+/// Specifies how a new asset will be provided on creating and updating
+/// Channels, Videos, Series and Person
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum NewAsset<ContentParameters> {
+    /// Upload to the storage system
+    Upload(ContentParameters),
+    /// A url string pointing at an asset
+    Uri(Vec<u8>),
```

Comment:
From chat: "we probably want to allow a vector of URIs, not just a single one, as a backup." - @bedeho 

Will update that in upcoming PR where I'm implementing the extrinsics

Reply:
Atlas could even be creative about it and do some nice amimated transitions between the multiple urls that gives some dynamic life to a view

Reply:
done in 59a7b3ca3fee7a7f73aac423d54a48183fd23772